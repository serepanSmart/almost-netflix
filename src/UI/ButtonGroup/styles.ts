import { IButtonGroup } from './index';
import styled, { css } from 'styled-components';

export const VerticalContainer = styled.div`
  margin: -5px;
  >* {
    margin: 5px;
  }
`;

interface IButtonGroupWrapper {
  styled: Pick<IButtonGroup, 'noWrap' | 'align' | 'vertical' | 'spacer'>;
}

export const Wrapper = styled.div<IButtonGroupWrapper>`
  display: ${({ styled: { align } }) => (align ? 'flex' : 'inline-flex')};
  align-items: flex-start;
  flex-wrap: ${({ styled: { noWrap } }) => !noWrap && 'wrap'};
  margin:
    ${({ styled: { spacer, vertical } }) => !spacer && !vertical && '-15px'};
  justify-content:
    ${({ styled: { align } }) => (align === 'right' ? 'flex-end' : align)};
  >* {
    margin: ${({ styled: { vertical } }) => !vertical && '15px'};
  }
  ${({ styled: { spacer, vertical } }) => spacer === 'top' && css`
    margin: ${() => (vertical ? '15px 0 0' : '10px -5px -5px')};
  `};
  ${({ styled: { spacer, vertical } }) => spacer === 'bottom' && css`
    margin: ${() => (vertical ? '0 0 15px' : '-5px -5px 10px')};
  `};
`;