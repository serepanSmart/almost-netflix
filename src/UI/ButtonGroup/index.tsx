import React from 'react';
import { VerticalContainer, Wrapper } from './styles';

export interface IButtonGroup {
  noWrap?: boolean;
  vertical?: boolean;
  align?: string;
  spacer?: string;
  children: React.ReactNode;
}

const ButtonGroup: React.FC<IButtonGroup> = ({
  noWrap,
  vertical,
  align,
  spacer,
  children,
  ...other
}) => (
  <Wrapper
    {...other}
    styled={{
      noWrap,
      align,
      spacer,
      vertical,
    }}
  >
    {vertical ? (
      <VerticalContainer>
        {children}
      </VerticalContainer>
    ) : children}
  </Wrapper>
);

export default ButtonGroup;