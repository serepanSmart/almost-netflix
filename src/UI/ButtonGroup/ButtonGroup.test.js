import { render } from '@testing-library/react';
import ButtonGroup from '.';

describe('Label component', () => {
  let component;
  beforeEach(() => {
    component = (props) => render(<ButtonGroup {...props} />);
  });

  test('should define ButtonGroup component', () => {
    expect(component()).toBeDefined();
  });

  test('should be rendered', () => {
    const { container } = component();
    expect(container.firstChild).toBeInTheDocument();
  });

  test('should match snapshot', () => {
    expect(component()).toMatchSnapshot();
  });

  test('should have "minus" margins', () => {
    const { container } = component({ vertical: true });
    const div = container.firstChild.querySelector('div');
    expect(div).toHaveStyle('margin: -5px');
  });

  test('should have additional space on the top', () => {
    const { container } = component({ spacer: 'top' });
    expect(container.firstChild).toHaveStyle('margin: 10px -5px -5px');
  });

  test('should have additional space on the bottom', () => {
    const { container } = component({ spacer: 'bottom' });
    expect(container.firstChild).toHaveStyle('margin: -5px -5px 10px');
  });
});
