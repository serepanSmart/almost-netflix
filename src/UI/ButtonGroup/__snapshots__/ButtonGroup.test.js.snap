// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: -15px;
}

.c0 >* {
  margin: 15px;
}

<body>
    <div>
      <div
        class="c0"
      />
    </div>
  </body>,
  "container": .c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: -15px;
}

.c0 >* {
  margin: 15px;
}

<div>
    <div
      class="c0"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
